build/circle-test.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -e
+
+outdir="${TMPDIR}"
+if [ "${CIRCLE_ARTIFACTS}" != "" ]; then
+    outdir="${CIRCLE_ARTIFACTS}"
+fi
+
+builder=$(dirname $0)/builder.sh
+
+# 1. Run "make check" to verify coding guidelines.
+echo "make check"
+time ${builder} make GITHOOKS= check | tee "${outdir}/check.log"; test ${PIPESTATUS[0]} -eq 0
+
+# 2. Verify that "go generate" was run.
+echo "verifying generated files"
+time ${builder} /bin/bash -c "(go generate ./... && git ls-files --modified --deleted --others --exclude-standard | diff /dev/null -) || (git add -A && git diff -u HEAD && false)" | tee "${outdir}/generate.log"; test ${PIPESTATUS[0]} -eq 0
+
+# 3. Run "make testrace".
+match='^panic|^[Gg]oroutine \d+|(read|write) by.*goroutine|DATA RACE'
+echo "make testrace"
+time ${builder} make GITHOOKS= testrace \
+     RACETIMEOUT=5m TESTFLAGS='-v --verbosity=1 --vmodule=monitor=2' | \
+    tr -d '\r' | tee "${outdir}/testrace.log" | \
```

Comment:
I think the `tr` should stay. At least locally it messes up the output. We need to see what causes it to be different on CI.